core/src/main/java/com/linecorp/armeria/client/WebClientRequestPreparation.java

```diff
@@ -66,11 +76,152 @@ public HttpResponse execute() {
         return client.execute(httpRequest, requestOptions);
     }
 
+    /**
+     * Sets the specified {@link ResponseAs} that converts the {@link HttpResponse} into another.
+     */
+    @UnstableApi
+    public <T> TransformingRequestPreparation<HttpResponse, T> as(ResponseAs<HttpResponse, T> responseAs) {
+        requireNonNull(responseAs, "responseAs");
+        return new TransformingRequestPreparation<>(this, responseAs);
+    }
+
+    /**
+     * Sets the specified {@link ResponseAs} that converts the {@link HttpResponse} into an
+     * {@link ResponseEntity}.
+     */
+    @UnstableApi
+    public <T> FutureTransformingRequestPreparation<ResponseEntity<T>> asEntity(
+            FutureResponseAs<ResponseEntity<T>> responseAs) {
+        requireNonNull(responseAs, "responseAs");
+        return new FutureTransformingRequestPreparation<>(this, responseAs);
+    }
+
+    /**
+     * Converts the content of the {@link HttpResponse} into bytes.
+     * For example:
+     * <pre>{@code
+     * WebClient client = WebClient.of("https://api.example.com");
+     * CompletableFuture<ResponseEntity<byte[]>> response =
+     *     client.prepare()
+     *           .get("/v1/items/1")
+     *           .asBytes()
+     *           .execute();
+     * }</pre>
+     */
+    @UnstableApi
+    public FutureTransformingRequestPreparation<ResponseEntity<byte[]>> asBytes() {
+        return asEntity(ResponseAs.bytes());
+    }
+
+    /**
+     * Converts the content of the {@link HttpResponse} into a {@link String}.
+     * For example:
+     * <pre>{@code
+     * WebClient client = WebClient.of("https://api.example.com");
+     * CompletableFuture<ResponseEntity<String>> response =
+     *     client.prepare()
+     *           .get("/v1/items/1")
+     *           .asString()
+     *           .execute();
+     * }</pre>
+     */
+    @UnstableApi
+    public FutureTransformingRequestPreparation<ResponseEntity<String>> asString() {
+        return asEntity(ResponseAs.string());
+    }
+
+    /**
+     * Converts the content of the {@link HttpResponse} into a {@link Path}.
+     * For example:
+     * <pre>{@code
+     * WebClient client = WebClient.of("https://api.example.com");
+     * CompletableFuture<ResponseEntity<Path>> response =
+     *     client.prepare()
+     *           .get("/v1/items/1")
+     *           .asPath(Paths.get("..."))
+     *           .execute();
+     * }</pre>
+     */
+    public FutureTransformingRequestPreparation<ResponseEntity<Path>> asPath(Path path) {
```

Comment:
Should we rename this to `asFile()`?

Reply:
Sounds good as `path` is used for a request path.